test/unit/transform/classes_test.py

```diff
@@ -303,6 +335,54 @@
     }
 )
 
+MIXED_TYPE_DATA: pd.DataFrame = pd.DataFrame(
+    columns=["id", "year", "report_date", "capacity_mw", "plant_name"],
+    data=[
+        (1, 1776, "2020-01-01T00:00:00.000000000", 132.1, "planty"),
+        (2, 1976, "2020-01-02", 2000.0, "genny"),
+        (3, 1976, date.today(), 123.1, "big burner"),
+        (4, 1976, "1985-02-01", 213.1, "sparky"),
+        (5, pd.NA, pd.NA, -5.0, "barry"),
+        (6, 2000, "2000-01-01", 231.1, "replace me"),
+        (7, pd.NA, pd.NA, 101.10, pd.NA),
+        (8, 2012, "01-01-2020", 899.98, "another plant name"),
+        (9, 1850, "2022-03-01T00:00:00.000000000", 543.21, np.nan),
+        (10, date.today().year, date.today(), 8.1, "cat corp"),
```

Comment:
Some other test cases that come to mind:

* Check that the type enforcement works by giving it e.g. a numeric string "123.456" as a fix to apply in a float column, and a number to apply in a string column (e.g. 123.456 should get turned into "123.456")
* Check that when an applied spot-fix cannot be cast to the data type of the input column it applies to, the appropriate exception is raised (e.g. give it a non-numeric string to apply in a float column).
* Check that when a given set of `idx_cols` criteria select more than one row, the spot fix is applied to all of the selected rows (e.g. try using a single spot fix to set all plant names with year=1976 to "Bicentennial")
* Check what happens when the spot fixes you give it select *zero* rows -- should result in no change to the input dataframe at all, right?

Are there any other edge cases you can think of that need to be exercised?

Reply:
> Check that the type enforcement works by giving it e.g. a numeric string "123.456" as a fix to apply in a float column, and a number to apply in a string column (e.g. 123.456 should get turned into "123.456")
Added.

> Check that when an applied spot-fix cannot be cast to the data type of the input column it applies to, the appropriate exception is raised (e.g. give it a non-numeric string to apply in a float column).
Added this and a check that the non-unique error is also raised as expected.

> Check that when a given set of idx_cols criteria select more than one row, the spot fix is applied to all of the selected rows (e.g. try using a single spot fix to set all plant names with year=1976 to "Bicentennial")
This should already be happening to convert all 1976 plants to have capacity_mw 999.9. One of these values is subsequently overwritten in a later spot fix, but two should remain changed.

> Check what happens when the spot fixes you give it select zero rows -- should result in no change to the input dataframe at all, right?
Added.

Reply:
I'm not seeing the string-to-numeric or numeric-to-string type casting tests. What am I missing?